overlord/snapstate/handlers.go

```diff
@@ -1159,24 +1159,32 @@ func maybeRestart(t *state.Task, info *snap.Info) {
 		return
 	}
 
-	// On a core system we may need a full reboot if
-	// core/base or the kernel changes.
-	if boot.ChangeRequiresReboot(info) {
-		t.Logf("Requested system restart.")
-		st.RequestRestart(state.RestartSystem)
-		return
-	}
-
 	// On core systems that use a base snap we need to restart
 	// snapd when the snapd snap changes.
 	model, err := ModelFromTask(t)
 	if err != nil {
 		logger.Noticef("cannot get model assertion: %v", model)
 		return
 	}
-	if model.Base() != "" && info.GetType() == snap.TypeSnapd {
+
+	bp, applicable := boot.Lookup(info, info.GetType(), model, release.OnClassic)
```

Comment:
this will return applicable false for snapd on classic but we do want to restart there, no?, missing a tests? me confused?

Reply:
if you're on classic you never make it this far down

Reply:
ah,  forgot that the code is fairly repetitive here.